From e3218cda9894bced0f64020b2fe211202009c2fa Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Fri, 13 Feb 2015 16:57:16 +0100
Subject: [PATCH 1/2] Replace cjson with json-c-0.12

:Release Notes:
Replace cjson with json-c.

:Detailed Notes:

:Testing Performed:

:QA Notes:

:Issues Addressed:
[DRD-3283] Replace cjson with json-c v0.12

Open-webOS-DCO-1.0-Signed-off-by: Martin Jansa <martin.jansa@lge.com>
Change-Id: Ib2d7aad4da16a78c7afddc50f8049b7b27780e23
---
 common/CMakeLists.txt | 8 ++++----
 imap/CMakeLists.txt   | 6 +++---
 pop/CMakeLists.txt    | 6 +++---
 smtp/CMakeLists.txt   | 6 +++---
 4 files changed, 13 insertions(+), 13 deletions(-)

diff --git a/common/CMakeLists.txt b/common/CMakeLists.txt
index 27c7f05..9ebdffa 100644
--- a/common/CMakeLists.txt
+++ b/common/CMakeLists.txt
@@ -42,9 +42,9 @@ pkg_check_modules(GLIB2 REQUIRED glib-2.0)
 include_directories(${GLIB2_INCLUDE_DIRS})
 webos_add_compiler_flags(ALL ${GLIB2_CFLAGS})
 
-pkg_check_modules(CJSON cjson)
-include_directories(${CJSON_INCLUDE_DIRS})
-webos_add_compiler_flags(ALL ${CJSON_CFLAGS})
+pkg_check_modules(JSON json-c)
+include_directories(${JSON_INCLUDE_DIRS})
+webos_add_compiler_flags(ALL ${JSON_CFLAGS})
 
 #pkg_check_modules(MOJODB mojodb)
 #include_directories(${MOJODB_INCLUDE_DIRS})
@@ -106,7 +106,7 @@ aux_source_directory(src/util util_files)
 
 add_library(email-common SHARED src/CommonErrors.cpp ${activity_files} ${client_files} ${data_files} ${exceptions_files} ${mimeparser_files} ${stream_files} ${async_files} ${email_files} ${glibcurl_files} ${network_files} ${util_files})
 
-target_link_libraries(email-common ${GLIB2_LDFLAGS} ${CJSON_LDFLAGS} ${PALMSOCKET_LDFLAGS} ${SANDBOX_LDFLAGS} ${PMLOG_LDFLAGS} ${LUNASERVICE_LDFLAGS} ${JEMALLOC_MT_LDFLAGS} ${ICU} ${ICUI18N} ${LIBZ} ${Boost_LIBRARIES} ${DB8_LDFLAGS} pthread curl)
+target_link_libraries(email-common ${GLIB2_LDFLAGS} ${JSON_LDFLAGS} ${PALMSOCKET_LDFLAGS} ${SANDBOX_LDFLAGS} ${PMLOG_LDFLAGS} ${LUNASERVICE_LDFLAGS} ${JEMALLOC_MT_LDFLAGS} ${ICU} ${ICUI18N} ${LIBZ} ${Boost_LIBRARIES} ${DB8_LDFLAGS} pthread curl)
 
 webos_build_pkgconfig()
 webos_build_library(NAME email-common)
diff --git a/imap/CMakeLists.txt b/imap/CMakeLists.txt
index 4bf2ba7..9ed3cc3 100644
--- a/imap/CMakeLists.txt
+++ b/imap/CMakeLists.txt
@@ -50,9 +50,9 @@ pkg_check_modules(LUNASERVICE REQUIRED luna-service2)
 include_directories(${LUNASERVICE_INCLUDE_DIRS})
 webos_add_compiler_flags(ALL ${LUNASERVICE_CFLAGS})
 
-pkg_check_modules(CJSON cjson)
-include_directories(${CJSON_INCLUDE_DIRS})
-webos_add_compiler_flags(ALL ${CJSON_CFLAGS})
+pkg_check_modules(JSON json-c)
+include_directories(${JSON_INCLUDE_DIRS})
+webos_add_compiler_flags(ALL ${JSON_CFLAGS})
 
 pkg_check_modules(EMAILCOMMON email-common)
 include_directories(${EMAILCOMMON_INCLUDE_DIRS})
diff --git a/pop/CMakeLists.txt b/pop/CMakeLists.txt
index b607deb..7a8dff9 100644
--- a/pop/CMakeLists.txt
+++ b/pop/CMakeLists.txt
@@ -50,9 +50,9 @@ pkg_check_modules(LUNASERVICE REQUIRED luna-service2)
 include_directories(${LUNASERVICE_INCLUDE_DIRS})
 webos_add_compiler_flags(ALL ${LUNASERVICE_CFLAGS})
 
-pkg_check_modules(CJSON cjson)
-include_directories(${CJSON_INCLUDE_DIRS})
-webos_add_compiler_flags(ALL ${CJSON_CFLAGS})
+pkg_check_modules(JSON json-c)
+include_directories(${JSON_INCLUDE_DIRS})
+webos_add_compiler_flags(ALL ${JSON_CFLAGS})
 
 #pkg_check_modules(MOJODB mojodb)
 #include_directories(${MOJODB_INCLUDE_DIRS})
diff --git a/smtp/CMakeLists.txt b/smtp/CMakeLists.txt
index e55c5c2..eea1645 100644
--- a/smtp/CMakeLists.txt
+++ b/smtp/CMakeLists.txt
@@ -50,9 +50,9 @@ pkg_check_modules(LUNASERVICE REQUIRED luna-service2)
 include_directories(${LUNASERVICE_INCLUDE_DIRS})
 webos_add_compiler_flags(ALL ${LUNASERVICE_CFLAGS})
 
-pkg_check_modules(CJSON cjson)
-include_directories(${CJSON_INCLUDE_DIRS})
-webos_add_compiler_flags(ALL ${CJSON_CFLAGS})
+pkg_check_modules(JSON json-c)
+include_directories(${JSON_INCLUDE_DIRS})
+webos_add_compiler_flags(ALL ${JSON_CFLAGS})
 
 #pkg_check_modules(MOJODB mojodb)
 #include_directories(${MOJODB_INCLUDE_DIRS})
-- 
2.8.4

